### PR TITLE
Gcov implementation

### PR DESCRIPTION
**Pull request by pordaniel01**
Gcov has been successfully installed. Coverage can be viewed now on Sonar-cloud.

### CODE REVIEW

**Comment by pordaniel01** [comment]
It seems gcov creates a successful code coverage report. Sonarcloud processes this data and showcases it:
https://sonarcloud.io/dashboard?id=osrep_Runaphys&pullRequest=29

Looks like the code has 100% coverage. However it seems like for some reason the sonar-scanner is not scanning the include header files, because it should evince minor code smells in the constants.h header, like it does on other branches. 
Sonar-cloud also displays the following warning message:

> Could not find ref 'development' in refs/heads, refs/remotes/upstream or refs/remotes/origin. You may see unexpected issues and changes. Please make sure to fetch this ref before pull request analysis.

I tried to remove sonar-clouds cache, but it didn't fix the issue. 
Otherwise everything is working.

**Comment by pordaniel01** [comment]
Another issue is, that CMake generates the required gcno and gcda files in the build directory, but gcov is looking for these files in the root. I tried to fixed this problem by writing a [gcov.sh](https://github.com/osrep/Runaphys/blob/gcov_implementation/gcov.sh) shell script, which moves these files to the root, where gcov can find them. This isn't an elegant solution, we should find something more convinient.

File: gcov.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#script that runs gcov coverage measurments
```
Comment (gergopokol) [inline gcov.sh:1]: This file looks like it deserves a review.

File: src/critical_field.cpp [hunk 2]
```diff
@@ -5,6 +5,8 @@
 #include "products.h"
 #include "cell.h"
 
+
```
Comment (gergopokol) [inline src/critical_field.cpp]: These inserted lines are to be removed.

File: src/growth_rate.cpp [hunk 3]
```diff
@@ -6,6 +6,7 @@
 #include "critical_field.h"
 #include "cell.h"
 
+
```
Comment (gergopokol) [inline src/growth_rate.cpp]: These inserted lines are to be removed.

File: src/critical_field.cpp (hunk 2 above)
Comment (gergopokol) [reply src/critical_field.cpp]: Done.

File: src/growth_rate.cpp (hunk 3 above)
Comment (gergopokol) [reply src/growth_rate.cpp]: Done.

**Review by gergopokol — APPROVED** [review]

**Pull request merged**